WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -485,10 +485,20 @@ private void resetFragments() {
     }
 
     private void moderateCommentOnActivityResult(Intent data) {
+
         Note note = NotificationsTable.getNoteById(StringUtils.notNullStr(data.getStringExtra
                 (NotificationsListFragment.NOTE_MODERATE_ID_EXTRA)));
+
+        if (note == null) {
```

Comment:
Yes I agree with you, the only way to get a NPE here is that the notes DB got a refresh in background (notes are refreshed at startup, and even later in background) while the details screen is on Foreground. Tapping on moderate cause the NPE.